plugins/merge-connection/src/index.ts

```diff
@@ -0,0 +1,541 @@
+        processedSheets++
+        await tick(
+          Math.min(90, Math.round(40 + (50 * processedSheets) / sheets.length)),
+          `Synced ${sheet.config.name}`
+        )
+      }
+
+      // Finally, update the lastSyncedAt date
+      await api.workbooks.update(workbookId, {
+        metadata: {
+          connections: [
+            {
+              ...workbook.metadata.connections[0],
+              lastSyncedAt: new Date().toISOString(),
+            },
+          ],
+        },
+      })
+      await tick(95, 'Updating workbook...')
+
+      return {
+        outcome: {
+          message: `${workbook.name} data has been successfully synced from ${workbook.name} to Merge.dev and from Merge.dev to Flatfile.`,
+        },
+      } as Flatfile.JobCompleteDetails
+    } catch (e) {
+      handleError(e, 'Error syncing connected workbook')
+    }
+  }
+}
+
+async function fetchAllSyncStatuses(
+  mergeClient: MergeClient,
+  category: string
+) {
+  let cursor: string | undefined
+  const allResults = []
+
+  do {
+    const paginatedSyncList = await mergeClient[category].syncStatus.list({
+      cursor,
+    })
+    allResults.push(...paginatedSyncList.results)
+    cursor = paginatedSyncList.next
+  } while (cursor)
+
+  return allResults
+}
+
+async function checkAllSyncsComplete(
+  mergeClient: MergeClient,
+  category: string
+) {
+  try {
+    const allSyncs = await fetchAllSyncStatuses(mergeClient, category)
+    const completedSyncs = allSyncs.filter(
+      (syncStatus) =>
+        syncStatus.status === Merge[category].SyncStatusStatusEnum.Done
+    ).length
+    const totalModels = allSyncs.length
+
+    return {
+      allComplete: completedSyncs === totalModels,
+      completedSyncs,
+      totalModels,
+    }
+  } catch (e) {
+    handleError(e, 'Error checking sync status')
+  }
+}
+
+async function waitForMergeSync(
+  mergeClient: MergeClient,
+  category: string,
+  tick: (progress?: number, message?: string) => Promise<Flatfile.JobResponse>
+): Promise<void> {
+  try {
+    let attempts = 0
+    let syncStatusComplete = false
+
+    while (attempts <= MAX_SYNC_ATTEMPTS && !syncStatusComplete) {
+      const { allComplete, completedSyncs, totalModels } =
+        await checkAllSyncsComplete(mergeClient, category)
+      syncStatusComplete = allComplete
+
+      await tick(
+        Math.min(40, Math.round(10 + (30 * completedSyncs) / totalModels)),
+        'Merge syncing with Integration...'
+      )
+
+      if (!syncStatusComplete) {
+        attempts++
+        console.log('Waiting for Merge to sync...')
+        await new Promise((resolve) =>
+          setTimeout(resolve, SYNC_RETRY_INTERVAL_MS)
+        )
+      }
+    }
+
+    if (!syncStatusComplete) {
+      throw new Error('Merge sync timed out')
+    }
+  } catch (e) {
+    handleError(e, 'Error waiting for Merge sync')
+  }
+}
+
+async function deleteSheetRecords(sheetId: string) {
+  try {
+    const { data: records } = await api.records.get(sheetId)
```

Comment:
does this fetch all the records from the sheet if there are a ton? I think the get records endpoint defaults to 1k. or it might be 10k in quickstore

Reply:
Good question. I had assumed it fetched all records...

Reply:
yeah, looks like for quickstore it will max out at 10k records. this should be okay for demo purposes, so maybe just add a comment?

we'd either need to create a bulk delete job and listen for it to complete _or_ I can try to get this `DELETE` endpoint to work with filters instead of just record ids https://github.com/FlatFilers/Platform/issues/5522

Reply:
Good to know. I'll add a todo comment.

Reply:
This will likely be changing from a delete/replace to and update.

Reply:
Comment: [3822ecb](https://github.com/FlatFilers/flatfile-plugins/pull/245/commits/3822ecb144ae883d1d41ce2fec318f0b26c89d69)